src/main/java/ch/njol/skript/expressions/ExprAdventureRestrictions.java

```diff
@@ -0,0 +1,160 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.BukkitUnsafe;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+@Name("Adventure Restrictions Of Item")
+@Description("Get or modify the adventure restrictions of an item.")
+@Examples({"add dirt to destroyable restrictions of player's tool",
+	"add (stone, diamond ore) to breakable keys of {_item}",
+	"clear breakable restrictions of {_item}",
+	"remove sand from destroyable keys of {_item}"})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class ExprAdventureRestrictions extends PropertyExpression<ItemType, ItemType> {
+
+	static {
+		if (Skript.methodExists(ItemMeta.class, "getDestroyableKeys")) {
+			Skript.registerExpression(ExprAdventureRestrictions.class, ItemType.class, ExpressionType.PROPERTY,
+				"[the] (break|destroy)able (key|block|item|restriction)s (of|on) %itemtypes%",
+				"[the] (placeable|build[able]) (key|block|item|restriction)s (of|on) %itemtypes%");
```

Comment:
I'm pretty happy with those honestly. I also wasnt a big fan of the patterns as they were, just wasnt too sure how to change them.

Maybe we could combine some of them and use a parse tag (which would result in 4 patterns instead of 8)? Or leave them as is for readability sake?

Reply:
I left it as 8 for readability, but if you can find a way to collapse them without sacrificing too much, go for it